main.go

```diff
@@ -197,6 +201,15 @@ func main() {
 
 	cacheRecorderReader := NewRestoreReader(cacheReader)
 
+	if bytesRead, err := ioutil.ReadAll(cacheRecorderReader); err == nil {
+		numBytesRead := len(bytesRead)
+		data := map[string]interface{}{
+			"archive_size": numBytesRead,
+		}
+		log.RInfof(stepID, "archive_size", data, "Size of cache archive: %d Bytes", numBytesRead)
+	}
+	cacheRecorderReader.Restore()
```

Comment:
RestoreReader can be restored only once: https://github.com/bitrise-steplib/steps-cache-pull/blob/master/reader.go#L10

If you call restore here subsequent restore will not work as expected: https://github.com/bitrise-steplib/steps-cache-pull/pull/19/files#diff-7ddfb3e035b42cd70649cc33393fe32cR224